fastlane/Fastfile

```diff
@@ -1,117 +1,341 @@
+    )
+
+  end
 
-    # Update the app identifier
-    update_app_identifier(
-      xcodeproj: "#{options[:project_name]}.xcodeproj",
-      plist_path: "#{options[:project_name]}/Info.plist",
-      app_identifier: options[:app_identifier]
+  desc "Creates the `App ID` and `Provisioning Profile` for the configuration mapped to `:testflight`."
+  lane :create_testflight_app do
+    
+    desc "Remember after this point to choose this profile in xCode Signing (Beta)"
+    create_app(
+      app_name: app_names[:testflight] % ENV['PROJECT_NAME'],
+      build_configuration: build_configurations[:testflight],
+      team_name: ENV['TEAM_NAME'],
+      skip_itc: false,
+      match_type: match_types[:testflight],
     )
 
-    # Download certificates
-    cert
-    # Download provisioning profiles
-    sigh
+  end
+
+  desc "Adds a new device and regenerates the `Provisioning Profile`s to include it."
+  lane :add_device do
 
-    # Build the app
+    desc "Ask the user for device name and device UUID"
+    device_name = prompt(text: 'Enter the device name: ')
+    device_udid = prompt(text: 'Enter the device UDID: ')
+    device_hash = {}
+    device_hash[device_name] = device_udid
+
+    desc "Register new device."
+    register_devices(
+      devices: device_hash
+    )
+
+    desc "Refresh provisioning profiles adding the new device."
+    match(
+      force_for_new_devices: true
+    )
+
+  end
+
+  desc "Builds the app creating the `.ipa` and `.dsym` files"
+  private_lane :build_app do |options|
+
+    desc "Download provisioning profiles"
+    match(
+      app_identifier: options[:app_identifier],
+      type: options[:match_type],
+      readonly: true
+    )
+
+    desc "Build the app using default project and scheme"
     gym(
-      scheme: "#{options[:scheme]}",
       configuration: options[:configuration],
-      codesigning_identity: options[:codesigning_identity],
+      include_symbols: true,
+      # bitcode is disabled for the dsym file to keep valid after app is uploaded to app store
+      # http://krausefx.com/blog/download-dsym-symbolication-files-from-itunes-connect-for-bitcode-ios-apps
+      include_bitcode: false
     )
 
   end
 
-  # Publish to Testflight
-  private_lane :publish_testflight do |options|
+  desc "Create App ID and Provisioning Profile in Member Center"
+  desc "Keep these new certificates and profiles in sync with Match repository"
+  private_lane :create_app do |options|
 
-    pilot
+    desc "Bundle identifier from xCode project"
+    bundle_identifier = read_project_property(
+      build_configuration: options[:build_configuration],
+      build_setting: 'PRODUCT_BUNDLE_IDENTIFIER'
+    )
+
+    desc "Create App ID in developer center"
+    produce(
+      app_name: options[:app_name],
+      app_identifier: bundle_identifier,
+      team_name: options[:team_name],
+      skip_itc: options[:skip_itc]
+    )
+
+    desc "Generate provisioning profile if no present"
```

Comment:
This won't "Generate" the profile because of the `readonly` property set to `false`. It'd be more correct to say it just `downloads` the profile. I don't recall finishing the discussion about this `readonly: false` but it's fine, I guess in the general case the `build_app` lane will be called times enough to get the provisioning profiles updated before publishing.